lua/agentic/session_manager.test.lua

```
@@ -699,6 +698,307 @@ describe("agentic.SessionManager", function()
+            get_instance_stub = spy.stub(AgentInstance, "get_instance")
+            get_instance_stub:invokes(function(provider_name, callback)
+                --- @type agentic.acp.ACPClient
+                local fake = {}
+                fake.state = "ready"
+                fake.provider_config = {
+                    name = provider_name or "Test",
+                    initial_model = nil,
+                    default_mode = nil,
+                }
+                fake.agent_info = {}
+                function fake:create_session(_h, cb)
```

Luacheck (1.2.0)

[warning] unused argument 'self'

(W212)